VarduhiGhukasyan/chapter_10_problem_2

```diff
@@ -0,0 +1,131 @@
+#include <iostream>
+#include <vector>
+#include <ctime>
+
+class NoteBook
+{
+public:
+
+    NoteBook() : page_count(0), cost(0), dirty_page_count(0) {}
+
+    NoteBook(int _page_count, int _cost) :page_count(_page_count), cost(_cost), dirty_page_count(0){}
+
+    int get_page_count() const { return page_count; }
+    int get_cost() const { return cost; }
+    int get_filled_page_count() const { return dirty_page_count; }
+
+    void set_page_count(int page_count)
+    {
+        this->page_count = (page_count >= 0) ? page_count : 0;
+    }
+
+    void set_cost(int cost)
+    {
+        this->cost = (cost >= 0) ? cost : 0;
+    }
+
+    void set_dirty_page_count(int dirty_page_count)
+    {
+        this->dirty_page_count = (dirty_page_count > 0 && dirty_page_count <= page_count) ? dirty_page_count : 0;
+    }
+
+    void print_notebook() const
+    {
+        std::cout << "Page_count " << page_count << "\n"
+            << "Cost " << cost << "$ \n"
+            << "Dirty_page_count " << dirty_page_count << "\n";
+
+    }
+
+    void set_(int page_count,int cost,int dirty_page_count)
+    {
+        this->page_count = (page_count>=0)?page_count:0;
+        this->cost = (cost >= 0) ? cost : 0;
+        this->dirty_page_count = (dirty_page_count > 0 && dirty_page_count <= page_count) ? dirty_page_count : 0;
+        
+    }
+
+    int clean_page() const
+    {
+        return page_count - dirty_page_count;
+    }
+
+    bool equal(const NoteBook&) const
+    {
+        if (this->page_count != page_count) return false;
+        if (this->cost != cost) return false;
+        if (this->dirty_page_count != page_count) return false;
+
+        return true;
```

Comment:
```suggestion
    bool equal(const NoteBook& other) const
    {
        if (this->page_count != other.page_count) return false;
        if (this->cost != other.cost) return false;
        if (this->dirty_page_count != other.page_count) return false;

        return true;
```

Reply:
One more possibility here is to override operator== with the exactly same syntax, but with better readability in usage.